src/d3a/models/state.py

```diff
@@ -56,17 +56,33 @@ def get_state(self) -> Dict:
 
     @abstractmethod
     def restore_state(self, state_dict: Dict):
-        pass
+        """Update the state of the device using the provided dictionary."""
 
     @abstractmethod
-    def delete_past_state_values(self, time_slot):
-        pass
+    def delete_past_state_values(self, current_time_slot):
+        """Delete the state of the device before the given time slot."""
 
     def __str__(self):
         return self.__class__.__name__
 
 
-class ConsumptionState(StateInterface):
+class ProsumptionInterface(StateInterface, ABC):
+    """Interface with common methods/variables shared by consumption and production devices."""
+
+    def __init__(self):
+        # Actual energy consumed/produced by the device at specific market slots
+        self._real_energy: Dict[DateTime, float] = {}
```

Comment:
I think the original reason for this structure is that we want to discourage having something like this in the strategies:
```python
self.state.real_energy.get(time_slot, 0.0)
```
Also, have a look at `set_desired_energy()`: in that case the setter is doing more than just setting a single variable. Doing it with public variables would be cumbersome and confusing in strategies, so we provide getters/setters to do it and to provide an interface that could be unified in the future.

Moreover, the State classes were already implemented and I thought it was better to be coherent rather than mixing up styles.